src/common/components/resetPasswordModal/resetPasswordModal.component.js

```diff
@@ -0,0 +1,500 @@
+    // This makes the process of creating an account more streamlined as we remove that click.
+    const verificationCodeButtonLink = document.querySelector('.button-link.enter-auth-code-instead-link')
+    verificationCodeButtonLink?.click()
+  }
+
+  injectErrorMessages (errors = this.signUpErrors) {
+    this.clearInjectedErrorMessages()
+
+    if (!errors?.length) {
+      return
+    }
+
+    // Inject error messages into the form since errors are cleared when switching steps/rerendering.
+    errors.forEach(error => {
+      const field = document.querySelector(`${inputFieldErrorSelectorPrefix}${error.property.replace(/\./g, '\\.')}`)
+      if (field) {
+        const errorElement = document.createElement('div')
+        errorElement.classList.add('okta-form-input-error', 'o-form-input-error', 'o-form-explain')
+        errorElement.setAttribute('role', 'alert')
+
+        const errorSummary = error.errorSummary
+        errorElement.textContent = Array.isArray(errorSummary) ? errorSummary.join(' ') : errorSummary
+        errorElement.prepend(createErrorIcon())
+
+        field.parentNode.classList.add('o-form-has-errors')
+        field.parentNode.appendChild(errorElement)
+      }
+    })
+
+    // Augment the error message box to specify which fields had errors. Since the email and
+    // password fields are on step 1 and the submission happens on step 3, the error messages for
+    // those fields won't be visible to the user.
+    if (this.currentStep === 4) {
+      // Skip the verification step because an invalid code comes back as "credentials.passcode" and
+      // to say that the password field was invalid would confuse the user
+      return
+    }
+
+    const errorBox = document.querySelector('.okta-form-infobox-error')
+    if (!errorBox) {
+      return
+    }
+
+    const errorFields = errors.map(error => {
+      if (error.property === 'userProfile.firstName') {
+        return this.translations.firstNameField
+      } else if (error.property === 'userProfile.lastName') {
+        return this.translations.lastNameField
+      } else if (error.property === 'userProfile.email') {
+        return this.translations.emailField
+      } else if (error.property === 'credentials.passcode') {
+        return this.translations.passwordField
+      }
+      return null
+    }).filter(Boolean)
+    if (!errorFields.length) {
+      return
+    }
+
+    // Remove the previous error fields
+    errorBox.querySelector('.error-fields')?.remove()
+
+    const errorMessage = document.createElement('p')
+    errorMessage.className = 'error-fields'
+    // The OKTA_SIGNUP_FIELDS_ERROR translation key is already loaded, so we can use
+    // $translate.instant to synchronously interpolate our fields into it
+    errorMessage.textContent = this.$translate.instant('OKTA_SIGNUP_FIELDS_ERROR', { fields: errorFields.join(', ') })
+    errorBox.append(errorMessage)
+  }
+
+  injectBackButton () {
```

Comment:
I wonder if we could add a help function that does these function ns where 2 or more files use.